cli/azd/pkg/input/console.go

```diff
@@ -495,10 +496,12 @@ func (c *AskerConsole) doInteraction(fn func(c *AskerConsole) error) error {
 		}()
 	}
 
-	if err := fn(c); err != nil {
-		return err
-	}
-	return nil
+	start := time.Now()
+	defer func() {
+		tracing.SetUsageAttributes(fields.PerfInteractTime.Int64(time.Since(start).Milliseconds()))
```

Comment:
If there is already a PerfInteractTime usage attribute in the baggage, this is going to overwrite the value instead of adding to it, correct?

In cases where we call `doInteraction` twice (e.g. in case where we prompt for a subscription and then a location, during an initial `provision`) are we going to lose the count of how much time the user spent in the first prompt?

Reply:
Good call. That is indeed a bug.

Reply:
I've switched the implementation to use a dedicated field here. We can maybe extend "Perf" type attributes as it's own category of tracing attributes that may be useful in the future, but currently it is coupled until we need to. The code for consumers does look a lot cleaner as a result:

https://github.com/Azure/azure-dev/blob/85f2eccd46ab7bea96adf99f5c01cbdc2a93d3b2/cli/azd/cmd/util.go#L302-L305